components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/AnalyticsPublisher.java

```diff
@@ -0,0 +1,273 @@
+	 *
+	 * @return the stream id of task instance stream
+	 * @throws MalformedStreamDefinitionException
+	 * @throws DifferentStreamDefinitionAlreadyDefinedException
+	 * @throws StreamDefinitionException
+	 * @throws AgentException
+	 */
+	private String getBPMNTaskInstanceStreamId() throws MalformedStreamDefinitionException,
+	                                                    DifferentStreamDefinitionAlreadyDefinedException,
+	                                                    StreamDefinitionException, AgentException {
+		StreamDefinition streamDefinition =
+				new StreamDefinition(AnalyticsPublisherConstants.TASK_STREAM_NAME,
+				                     AnalyticsPublisherConstants.STREAM_VERSION);
+		streamDefinition.setDescription(AnalyticsPublisherConstants.TASK_STREAM_DESCRIPTION);
+		streamDefinition.addPayloadData(AnalyticsPublisherConstants.TASK_DEFINITION_ID,
+		                                AttributeType.STRING);
+		streamDefinition
+				.addPayloadData(AnalyticsPublisherConstants.TASK_INSTANCE_ID, AttributeType.STRING);
+		streamDefinition
+				.addPayloadData(AnalyticsPublisherConstants.START_TIME, AttributeType.STRING);
+		streamDefinition.addPayloadData(AnalyticsPublisherConstants.END_TIME, AttributeType.STRING);
+		streamDefinition.addPayloadData(AnalyticsPublisherConstants.DURATION, AttributeType.LONG);
+		streamDefinition.addPayloadData(AnalyticsPublisherConstants.ASSIGNEE, AttributeType.STRING);
+		return dataPublisher.defineStream(streamDefinition);
+	}
+
+	/**
+	 * Create a data publisher to publish the data as events
+	 *
+	 * @return DataPublisher object
+	 * @throws MalformedURLException
+	 * @throws AgentException
+	 * @throws AuthenticationException
+	 * @throws TransportException
+	 */
+	private DataPublisher createDataPublisher()
+			throws MalformedURLException, AgentException, AuthenticationException,
+			       TransportException, UserStoreException, RegistryException {
+		DataPublisher dataPublisher = new DataPublisher(BPMNDataReceiverConfig.getThriftURL(),
+		                                                BPMNDataReceiverConfig.getUserName(),
+		                                                BPMNDataReceiverConfig.getPassword());
+
+		return dataPublisher;
+	}
+
+	/**
+	 * Stop the data publisher
+	 */
+	private void stopDataPublisher() {
+		dataPublisher.stop();
+	}
+
+	/**
+	 * Get meta data of the instances to publish them.
+	 *
+	 * @return new object array
+	 */
+	private Object[] getMeta() {
+		return new Object[] {};
+	}
+
+	/**
+	 * Set trust store parameters as the system property for encryption
+	 */
+	private void setTrustStoreParams() {
+		File filePath = new File("src/main/resources");
+		if (!filePath.exists()) {
+			filePath = new File("resources");
+		}
+		String trustStore = filePath.getAbsolutePath();
+		System.setProperty("javax.net.ssl.trustStore", trustStore + "/client-truststore.jks");
+		System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
+	}
```

Comment:
Better to move string literals to a constants file (e.g. "src/main/resources", "client-truststore.jks").